docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,127 @@
+
+As a fallback mechanism, when using manual sync with the force option, Argo CD will not perform dependency resolution and the sync will proceed regardless of the Application's dependencies' states.
+
+## Feature state and caveats
+
+This feature is in [beta state](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md#beta). It will be polished across the upcoming releases.
+
+Bug reports and enhancement requests are most welcome.
+
+## Limitations and known issues
+
+A list of things that are good to know before getting started with Application dependencies:
+
+* An Application can only depend on other Applications that use the same AppProject and are in the same namespace as the dependent Application.
+* A circular dependency defined anywhere in the dependency chain is considered to be an error and will prevent any application in the chain from syncing.
+* Syncing an application that is dependent on other apps will not trigger a sync of the dependencies. However, manual sync of a dependent application whose dependencies are not satisfied will block unless force sync is used.
+* Dependencies currently can only be configured in a declarative manner; the UI and CLI do not yet have first class support for configuring dependencies.
+
+## Configuring Application dependencies
+
+The dependencies of an Application are configured using the field `dependsOn` in the Application's spec. The field `dependsOn` is a complex data type, which lets you define one or more `selectors` to match a set of other Applications that this Application will be dependent upon.
+
+Each selector can match Applications according to one or more properties. If you specify more than one matcher in a selector, each matcher has to match in order for an Application to be considered a dependency. For example, if you specify a `labelSelector` and a `namePattern`, both will have to match in order for an Application to be selected.
+
+Additionally, the behaviour of dependency resolution can be configured on a per Application basis.
+
+## Dependency selectors
+
+Currently, the available properties to match against are `labels` and the `name` of the Application. Each selector can yield zero or more applications.
+
+### Matching on labels
+
+It is possible and recommended to use a standard label selector to select the dependencies for your Application:
+
+```yaml
+spec:
+  dependsOn:
+    selectors:
+    - labelSelector:
+        matchLabels:
+            name: some-app
+```
+
+The above YAML will have the Application depend on any other Application that matches the given label selector. In this case, any application carrying a label `name` with the value of `some-app`.
+
+### Matching on name patterns
+
+Dependencies can be matched on the name of Applications by specifying the `namePattern` selector. The `namePattern` selector takes a list of Application name patterns to match against the Applications' `.metadata.name` property. Specifying multiple patterns 
```

Comment:
```suggestion
Dependencies can be matched on the name of Applications by specifying the `namePattern` selector. The `namePattern` selector takes a list of Application name patterns to match against the Applications' `.metadata.name` property. Specifying multiple patterns:
```

It feels like there are words / whitespace missing near `Specifying multiple patterns`